icaruscode/IcarusObj/PMTBeamSignal.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file   icaruscode/IcarusObj/PMTBeamSignal.h
+ * @brief  Holds the event-by-event RWM or EW times
+ * @author Matteo Vicenzi (mvicenzi@bnl.gov)
+ * @date   March 14 2024
+ */
+
+#ifndef ICARUSCODE_ICARUSOBJ_PMTBEAMSIGNAL_H
+#define ICARUSCODE_ICARUSOBJ_PMTBEAMSIGNAL_H
+
+// C/C++ standard libraries
+#include <limits>
+
+namespace icarus::timing{
+		
+        /// Special value to denote no special channel information
+        static constexpr auto NoChannel = std::numeric_limits<unsigned int>::max();
+        /// Special value to denote no time channel information
+        static constexpr double NoTime = 0.;
+        /// Special value to denote no sample information
+        static constexpr size_t NoSample = 0;
+
+	struct PMTBeamSignal {
+		
+		/// Special channel this time was extracted from
+		unsigned int specialChannel = NoChannel;
+	 	
+                /// Board on which the special channel is on
+                std::string digitizerLabel = "";    	
+	 	
+                /// Crate this time applies to
+                std::string crate = "";    
+		
+                /// Sample within the waveform where the reference signal is found
+		size_t sample = NoSample;
+		
+                /// Start time in electronics time [us]
+		double startTimeAbs = NoTime;
+		
+                /// Start time relative to trigger time [us]
+		double startTime = NoTime;
+		
+                PMTBeamSignal(unsigned int ch, std::string b, std::string c,
+			      size_t s, double t, double tt):
+			     specialChannel(ch), digitizerLabel(b), crate(c), sample(s),
+			     startTimeAbs(t), startTime(tt) {};
+                
+                PMTBeamSignal(){};
+
+		/// Returns whether the time is valid.
+		bool isValid() const { return ((sample != NoSample) && (startTime != NoTime)); }
```

Comment:
Major concern here: `NoTime` is `0`, and `0` µs is a valid, albeit questionable, value for `startTime`: supposedly, `startTime` should be consistently larger than 0 by definition, but an algorithm _might_ end up with `0` as a value.
The definition of validity here makes a strange effect where an algorithm returning `startTime` of 0.0 would make the object invalid, while returning 0.1 , or -0.1, microseconds would leave it valid.
I suggest to change the implementation of `isValid()` to either check only `sample`, or use the relatively safer `startTimeAbs`, or to change the value of  `NoTime` to something inherently safe.

Reply:
I agree with you, and I can't fully remember why I opted for this.
It maybe comes from before I implemented a validity check, and I wanted to make the least amount of damage to other times (subtracting a zero) in case the signal was missing for that crate.

I'm going to change it to `std::numeric_limits<double>::max()`.